### PR TITLE
Flash: Added status register telecommand

### LINKED ISSUE #149: Flash Telecommand: get and print the Status Register for a flash module
Super easy command. Would love to get it in `main`. Would be helpful for checking on the general status of flash memory modules during a variety of testing actions.

Would this implementation be the same or different for the NAND vs. NOR modules, @Saksham-P?

### PR DESCRIPTION
**Pull request by KaleF07**
Resolves #149 

**Summary**
Added a simple telecommand that reads the status register value from the flash memory module and prints as hex value.

### CODE REVIEW

**Comment by DeflateAwning** [comment]
Feel free to add the "pr-ready-for-review" tag when this is ready for testing/review!

**Comment by KaleF07** [comment]
I added another telecommand for testing purposes, however it will not be necessary to include afterwards

**Comment by NuclearTea** [comment]
@KaleF07 Just to make sure, you have tested this yourself right?

**Comment by KaleF07** [comment]
> @KaleF07 Just to make sure, you have tested this yourself right?

Yes, I have tested `flash_read_status_register()`. To help with testing purposes, I tested setting WEL (2nd bit of the SR) to view the change.

Using `flash_read_status_register()` before enabling WEL:

image

After enabling WEL: 

image

I hope this helps!

File: firmware/Core/Src/telecommands/flash_telecommand_defs.c [hunk 1]
```diff
@@ -263,7 +263,7 @@ uint8_t TCMDEXEC_flash_erase(const char *args_str, TCMD_TelecommandChannel_enum_
     snprintf(
         &response_output_buf[strlen(response_output_buf)],
         response_output_buf_len - strlen(response_output_buf) - 1,
-        " Successfully erased page at address %lu on chip %d.\n", 
+        "Successfully erased block containing page address %lu on chip %d.\n",
```
Comment (DeflateAwning) [inline firmware/Core/Src/telecommands/flash_telecommand_defs.c:266]: "Page address" or "page number"? A page address would be the full address of the first byte of the page, whereas the page number would be `page_address / page_size`. 

Double-check that the variable name matches the description.

File: firmware/Core/Src/telecommands/flash_telecommand_defs.c [hunk 2]
```diff
@@ -345,5 +345,86 @@ uint8_t TCMDEXEC_flash_reset(const char *args_str, TCMD_TelecommandChannel_enum_
...
+
+    if (chip_num_u64 >= FLASH_NUMBER_OF_FLASH_DEVICES) {
+        snprintf(
+            response_output_buf, response_output_buf_len,
+            "Chip number is out of range. Must be 0 to %d.",
+            FLASH_NUMBER_OF_FLASH_DEVICES - 1);
+        return 2;
+    }
+
+    uint8_t chip_num = (uint8_t)chip_num_u64;
```
Comment (DeflateAwning) [inline firmware/Core/Src/telecommands/flash_telecommand_defs.c:376]: Make `const`

File: firmware/Core/Src/telecommands/flash_telecommand_defs.c [hunk 3]
```diff
@@ -345,5 +345,86 @@ uint8_t TCMDEXEC_flash_reset(const char *args_str, TCMD_TelecommandChannel_enum_
...
+    uint8_t status_reg_val;
+    const FLASH_error_enum_t comms_err = FLASH_read_status_register(&hspi1, chip_num, &status_reg_val);
+    if (comms_err != 0) {
+        snprintf(
+            response_output_buf, response_output_buf_len,
+            "Error reading status register: %d",comms_err);
+            return 2;
```
Comment (DeflateAwning) [inline firmware/Core/Src/telecommands/flash_telecommand_defs.c:383]: Check indentation.

File: firmware/Core/Src/telecommands/flash_telecommand_defs.c [hunk 4]
```diff
@@ -345,5 +345,86 @@ uint8_t TCMDEXEC_flash_reset(const char *args_str, TCMD_TelecommandChannel_enum_
...
+        return 2;
+    }
+
+    uint8_t chip_num = (uint8_t)chip_num_u64;
+    const FLASH_error_enum_t comms_err = FLASH_write_enable(&hspi1, chip_num);
+    if (comms_err != 0) {
+        snprintf(
+            response_output_buf, response_output_buf_len,
+            "Error enabling write latch: %d",comms_err);
+            return 2;
```
Comment (DeflateAwning) [inline firmware/Core/Src/telecommands/flash_telecommand_defs.c:426]: Indentation

**Review by DeflateAwning — CHANGES_REQUESTED** [review]
Looks awesome! Nice and easy.

Couple tiny changes, but no real issues.

File: firmware/Core/Src/telecommands/flash_telecommand_defs.c (hunk 1 above)
Comment (KaleF07) [reply firmware/Core/Src/telecommands/flash_telecommand_defs.c:266]: you're right, the proper description should be page number. I'll also changed the name of the variable to properly match this description.

**Review by DeflateAwning — APPROVED** [review]
Thanks!

**Pull request merged**